luaparse.js

```diff
@@ -889,6 +889,52 @@
     };
   }
 
+  function scanLuaJITImaginaryUnitSuffix() {
+    if (options.luaVersion !== 'LuaJIT') return;
+
+    // Imaginary unit number suffix is optional.
+    // See http://luajit.org/ext_ffi_api.html#literals
+    if ('iI'.indexOf(input.charAt(index) || null) >= 0) {
+      ++index;
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  function scanLuaJITInt64Suffix() {
+    if (options.luaVersion !== 'LuaJIT') return;
+
+    // Int64/uint64 number suffix is optional.
+    // See http://luajit.org/ext_ffi_api.html#literals
+
+    if ('uU'.indexOf(input.charAt(index) || null) >= 0) {
+      ++index;
+      if ('lL'.indexOf(input.charAt(index) || null) >= 0) {
+        ++index;
+        if ('lL'.indexOf(input.charAt(index) || null) >= 0) {
+          ++index;
+          return true;
+        } else {
+          // UL but no L
+          raise({}, errors.malformedNumber, input.slice(tokenStart, index));
```

Comment:
Maybe the function could return the prefix encountered, if any; this information could be then attached to the final AST node.

Reply:
Plus, the triply nested `if` statements look rather ugly.